drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3341,7 +3373,9 @@ static int rswitch_ndev_create(struct rswitch_private *priv, int index)
 	INIT_LIST_HEAD(&rdev->tc_u32_list);
 	INIT_LIST_HEAD(&rdev->tc_matchall_list);
 	INIT_LIST_HEAD(&rdev->tc_flower_list);
-	priv->rdev[index] = rdev;
+	INIT_LIST_HEAD(&rdev->list);
```

Comment:
It's not a necessary operation when you just add entry to list, so it can be omitted.